### PR TITLE
Update Go version to 1.16 for alpine image

### PR DESCRIPTION
**Pull request by MrWormHole**
## Description
Had to bump the versions over 1.15 due to arm64 bug from golang alpine image

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md)) https://github.com/openfaas/templates/issues/268

## Which issue(s) this PR fixes 

Fixes # The issue fixes arm64 faas-cli up which doesn't work on golang alpine image 1.15. It works with 1.16 and 1.17 tho.

## How Has This Been Tested?

Get raspberry pi 4/400, install faas-cli, change templates dockerfile to arm64. Do faas-cli up and you will get error about amd64 image tried to fetch git in arm64 system.

This will affect all ARM64 users who want to do deployments with faas-cli.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Version change (see: Impact to existing users)

## Impact to existing users
If you are using arm64 system, you won't be able to do faas-cli up -f stacks.yml, it requires a change in dockerfile and after that is done to linux/arm64 somehow golang alpine image 1.15 still installs amd64 which I think the related issue to docker repository. Official Alpine go images are 1.16 and 1.17 now. 1.15 is no longer supported for arm64.

1 15-alpine-go-image-issue

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: template/go/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 FROM --platform=${TARGETPLATFORM:-linux/amd64} ghcr.io/openfaas/classic-watchdog:0.1.5 as watchdog
-FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.15-alpine3.13 as builder
+FROM --platform=${BUILDPLATFORM:-linux/amd64} golang:1.17-alpine3.13 as builder
```
Comment (alexellis) [inline template/go/Dockerfile:2]: We would be able to sign off on Go 1.16, but 1.17 is untested.

A PR would be needed for this repo also https://github.com/openfaas/golang-http-template

**Review by alexellis — CHANGES_REQUESTED** [review]
Please change this to 1.16.

File: template/go/Dockerfile (hunk 1 above)
Comment (MrWormHole) [reply template/go/Dockerfile:2]: Perfect, That's the template I have been working on for a PoC serverless API

**Comment by MrWormHole** [comment]
> Please change this to 1.16.

Done btw

File: template/go/Dockerfile [hunk 2]
```diff
@@ -1,63 +1,63 @@
...
-
-USER app
-
-ENV fprocess="./handler"
-EXPOSE 8080
-
-HEALTHCHECK --interval=3s CMD [ -e /tmp/.lock ] || exit 1
-
-CMD ["./fwatchdog"]
+FROM --platform=${TARGETPLATFORM:-linux/amd64} ghcr.io/openfaas/classic-watchdog:0.1.5 as watchdog
```
Comment (alexellis) [inline template/go/Dockerfile:1]: Why have so many lines changed?

Comment (MrWormHole) [reply template/go/Dockerfile:1]: I don't know, it is essentially all same, my assumption is CR LF and LF stuff on that file creates git to do this

Comment (MrWormHole) [reply template/go/Dockerfile:1]: https://stackoverflow.com/questions/19593909/git-diff-sees-whole-file-as-changed-when-its-not

Comment (alexellis) [reply template/go/Dockerfile:1]: Are you a Windows user primarily?

**Review by alexellis — APPROVED** [review]
Approved

**Comment by alexellis** [comment]
Sorry for the delay here, the PR is appreciated and please do keep sending them.

Approved to run CI, please can you rebase?

**Comment by MrWormHole** [comment]
> Sorry for the delay here, the PR is appreciated and please do keep sending them.
> 
> Approved to run CI, please can you rebase?

There is a merge conflict so I think, it is not possible to do rebase anymore

**Comment by alexellis** [comment]
You can rebase, but it looks like you've introduced unnecessary merge commits, which we do not want.

The easiest course of action for you is to reset these commits and apply them again after synchronising your branch.

https://git-scm.com/docs/git-reset

**Comment by MrWormHole** [comment]
The CI/CD pipeline doesn't look good at the moment due to last change. I will just close and re-open this issue. I am very much against rebasing, it doesn't solve it when I have a merge conflict in a file.

**Comment by MrWormHole** [comment]
Closing due to aging

**Pull request closed without merging**